components/PlantForm.js

```diff
@@ -63,28 +67,44 @@ const fertiliserOptions = [
   { id: "fertiliserSeason4", value: "Winter", label: "Winter" },
 ];
 
-export default function PlantForm({ onCreatePlant }) {
+export default function PlantForm({
+    onCreatePlant,
+    onEditPlant,
+    isEditMode = false,
+    initialData = {}
```

Comment:
- Consider relying on `isFormVisible` from parent for edit mode instead of isEditMode
-  reeceive `onCancel` function to allow form to close itself or in clicking cancel.

```js
    onCreatePlant,
    onEditPlant,
    initialData = {}
    isFormVisible= false,
    onCancel
```